tests/test_auto_offer_on_device_rest.py

```diff
@@ -43,24 +53,23 @@ def on_trade(self, trade_info):
     def on_finish(self, finish_info):
         self.is_finished = True
 
-number_of_areas = 10
-load_names = [str(sys.argv[2]) if i == 0 else str(sys.argv[2])+' '+str(i+1) for i in range(number_of_areas)]
-pv_names = [str(sys.argv[3]) if i == 0 else str(sys.argv[3])+' '+str(i+1) for i in range(number_of_areas)]
-connected_devices = []
-
-for i in range(number_of_areas):
+for i in range(len(args.load_names)):
     # Connects one client to the load device
     load = AutoOfferBidOnMarket(
-        simulation_id= str(sys.argv[1]), 
-        device_id= get_area_uuid_from_area_name_and_collaboration_id(str(sys.argv[1]), load_names[i], 'https://d3aweb-dev.gridsingularity.com'),
+        simulation_id= args.simulation_id, 
+        device_id= get_area_uuid_from_area_name_and_collaboration_id(
+            args.simulation_id, args.load_names[i], 
+            'https://d3aweb-dev.gridsingularity.com'),
         domain_name='https://d3aweb-dev.gridsingularity.com',
         websockets_domain_name='wss://d3aweb-dev.gridsingularity.com/external-ws',
         autoregister=True)
     connected_devices.append(load)
     # Connects a second client to the pv device
     pv = AutoOfferBidOnMarket(
```

Comment:
Here, too. If you follow the following suggestion, then you can even parse lists that have different lengths for pv and lodas.
```suggestion
for pv in args.pv_names:
    pv = AutoOfferBidOnMarket(
```

Reply:
Yes, changing it as well, thanks